modules/statics/src/account.ts

```diff
@@ -138,6 +142,23 @@ export class StellarCoin extends AccountCoinToken {
   }
 }
 
+/**
+ * The Hedera coin needs a client set with the node account Id.
+ * It's an account based coin that needs the node account ID
+ * where the transaction will be sent.
+ */
+export class HederaCoin extends AccountCoinToken {
```

Comment:
If I understand the question correctly, using `implements` asks me to redefine each property for AccountCoinToken and then initialize those in the constructor. Using extends allows me to use the `super` in the constructor to send the remaining parameters. 
About HederaCoinConstructorOptions that is an interface but used for the constructor parameters only. It's very similar to StellarCoinConstructorOptions and Erc20ConstructorOptions but the naming is different so it better represents the value that we need. Also, that way it could be extended with more hedera specific properties if we need it.
Was that the question?

Reply:
so yeah basically you should be able to implement the two interfaces over a class:

X implements IAccountCoinToken, IHederaCoinConstructorOptions

Reply:
ah you know what. we are not setup to do things like this since AccountCoinToken is a class